internal/dataplane/parser/golden_test.go

```diff
@@ -0,0 +1,300 @@
+	filesInDirectory, err := os.ReadDir(path)
+	require.NoError(t, err, "failed to iterate over files in test case directory %s", path)
+
+	for _, fileInDirectory := range filesInDirectory {
+		// First, let's skip the files we want to keep.
+		if fileInDirectory.Name() == inFileName || strings.HasSuffix(fileInDirectory.Name(), settingsFileSuffix) {
+			continue
+		}
+
+		// Then, let's remove any other files.
+		err = os.Remove(filepath.Join(path, fileInDirectory.Name()))
+		require.NoError(t, err, "failed to remove file %s", filepath.Join(path, fileInDirectory.Name()))
+	}
+}
+
+// resolveSetsOfParserSettingsForTestCaseDir returns a slice of parserSettings, each of which represents a combination of
+// feature flags and Kong version.
+// The function iterates over a test case directory containing zero or more files named "<name>_settings.yaml".
+// If it doesn't find any settings files, it returns a single parserSettings with default feature flags and Kong version.
+func resolveSetsOfParserSettingsForTestCaseDir(t *testing.T, path string) []parserSettings {
+	// Iterate over all files in the directory and look for settings files.
+	files, err := os.ReadDir(path)
+	require.NoErrorf(t, err, "failed to iterate over files in test case directory %s", path)
+
+	setsOfParserSettings := []parserSettings{
+		// Always include a parserSettings with default feature flags and Kong version.
+		{
+			name:         "default",
+			featureFlags: defaultFeatureFlags(),
+			kongVersion:  defaultKongVersion,
+		},
+	}
+
+	// Iterate over all settings files and create a parserSettings for each.
+	for _, file := range files {
+		require.False(t, file.IsDir(), "unexpected directory %s in test case directory %s", file.Name(), path)
+
+		// Skip files that are not settings files.
+		if !strings.HasSuffix(file.Name(), settingsFileSuffix) {
+			continue
+		}
+
+		require.NotEqual(t, "default_settings.yaml", file.Name(),
+			"settings file name must not be default_settings.yaml - it's reserved for the default settings")
+
+		// Load the settings file and use it.
+		settingsFile := filepath.Join(path, file.Name())
+		setsOfParserSettings = append(setsOfParserSettings, unmarshalSettingsFile(t, settingsFile))
+	}
+
+	return setsOfParserSettings
+}
+
+type parserSettings struct {
+	name         string
+	featureFlags parser.FeatureFlags
+	kongVersion  semver.Version
+}
+
+// unmarshalSettingsFile unmarshals a settings file and returns a parserSettings struct.
+// All feature flags and Kong version specified in the settings file will be used to override the defaults.
+func unmarshalSettingsFile(t *testing.T, path string) parserSettings {
+	// It specifies only the json tags, because we're using "sigs.k8s.io/yaml" to unmarshal the file and that
+	// package respects only json tags: "Unmarshal converts YAML to JSON then uses JSON to unmarshal into an object".
+	type settingsFile struct {
+		FeatureFlags map[string]bool `json:"feature_flags"`
```

Comment:
I think that at some point we could consider using a `struct` instead of `type FeatureGates map[string]bool` with named fields for feature flags. But then Unmarshalling and Marshalling could be a challege 🤔 but that's just some food for thought.